opl2test: adapt for Linux

vincentbernat opened the pull request:
This can be compiled with:

    g++ -g -Wall main.cpp OPL2.cpp demotune.cpp timer.cpp -o opl2test

For some reason, this doesn't sound exactly like it should (for
example, on the YT video from the 8-bit guy or when running inside
VirtualBox). The `NOTE_C - 1` is suspicious. However, using 11 instead
doesn't change the result much.

I have tried another version using timers to better match the DOS
version, but this doesn't help on this side. Tempo is correct. I lack
words to describe the difference. ;-)

Here is my output: [output.ogg.zip](https://github.com/pdewacht/adlipt/files/1721859/output.ogg.zip) (I had to zip it to be able to upload)

Of course, this is not to be merged (as this breaks DOS support).

vincentbernat committed:
opl2test: adapt for Linux

This can be compiled with:

    g++ -g -Wall main.cpp OPL2.cpp demotune.cpp timer.cpp -o opl2test

This doesn't use timers, unlike the DOS version but just sleep a bit
instead of halting. This seems good enough for the test.

pdewacht commented:
There was a bit of code that assumed `char` was unsigned. Can you rebase and try again?

vincentbernat commented:
Thanks! It now works just fine.

[output.ogg.zip](https://github.com/pdewacht/adlipt/files/1725416/output.ogg.zip)

vincentbernat committed:
opl2test: restore DOS compatibility

There is nothing common between DOS and Linux for timers and main
program, so just use separate files. Throw some #ifdef for OPL2.cpp.

Not tested for DOS.

vincentbernat commented:
I have pushed another commit which should restore DOS support (not tested). Feel free to *not* merge it if you feel this is out of scope (or too ugly).

vincentbernat committed:
opl2test: use libieee1284 to control parallel port

vincentbernat commented:
Add a last commit to use libieee1284. This would make the code a bit more OS-independant (but I still use `__linux__`).

vincentbernat commented:
FI, I have also made patches for ScummVM and QEMU:
 - https://github.com/scummvm/scummvm/pull/1137
 - https://patchwork.ozlabs.org/patch/874876/

pdewacht commented:
Neat!

I've merged this pull request, but I think I'm going to hack on it a bit later to get rid of the ifdefs.
